### PR TITLE
#374: Add local registry support

### LINKED ISSUE #374: Add local registries to quickstart install
As per @evankanderson 's comment on [slack](https://knative.slack.com/archives/C93E33SN8/p1673043271952749), it'd be good to add local registries to the quickstart installs:

Kind: https://kind.sigs.k8s.io/docs/user/local-registry/

Minikube: https://minikube.sigs.k8s.io/docs/handbook/registry/#enabling-insecure-registries

/good-first-issue

### PR DESCRIPTION
**Pull request by ehudyonasi**
# Changes

-
-
-

/kind <kind>

Fixes #374 

**Release Note**

```release-note
Quickstart will now create a local registry. 
```

**Docs**

```docs

```

### CODE REVIEW

File: pkg/minikube/minikube.go [hunk 1]
```diff
@@ -203,7 +203,8 @@ func createNewCluster() error {
 		"--cpus", cpus,
 		"--memory", memory,
 		"--profile", clusterName,
-		"--wait", "all")
+		"--wait", "all",
+		"--insecure-registry", "10.0.0.0/24")
```
[the remark this answers is not in the document]
Comment (ehudyonasi) [reply pkg/minikube/minikube.go]: Sure, added.

File: pkg/kind/kind.go [hunk 2]
```diff
@@ -110,6 +115,46 @@ func checkDocker() error {
...
+	deleteContainerRegistry := deleteContainerRegistry()
+	if err := deleteContainerRegistry.Run(); err != nil {
+		return fmt.Errorf("failed to delete local registry: %w", err)
+	}
+	localRegCheck := exec.Command(
+		"docker", "run", "-d", "--restart=always", "-p", "127.0.0.1:"+container_reg_port+":5000",
+		"--name", container_reg_name, "registry:2",
+	)
+	if err := localRegCheck.Run(); err != nil {
+		return fmt.Errorf("failed to create local registry container: %s", err.Error())
```
Comment (psschwei) [inline pkg/kind/kind.go:128]:
```suggestion
		return fmt.Errorf("failed to create local registry container: %w", err.Error())
```

to get the lint action passing

File: pkg/kind/kind.go [hunk 3]
```diff
@@ -110,6 +115,46 @@ func checkDocker() error {
...
+	deleteContainerRegistry := deleteContainerRegistry()
+	if err := deleteContainerRegistry.Run(); err != nil {
+		return fmt.Errorf("failed to delete local registry: %w", err)
+	}
+	localRegCheck := exec.Command(
+		"docker", "run", "-d", "--restart=always", "-p", "127.0.0.1:"+container_reg_port+":5000",
+		"--name", container_reg_name, "registry:2",
+	)
+	if err := localRegCheck.Run(); err != nil {
+		return fmt.Errorf("failed to create local registry container: %w", err.Error())
```
Comment (psschwei) [inline pkg/kind/kind.go:128]: sorry, missed the bit on my previous review

```suggestion
		return fmt.Errorf("failed to create local registry container: %w", err)
```

**Review by psschwei — COMMENTED** [review]
Thanks for adding registry support!

/lgtm
/approve

**Pull request merged**